central/processlisteningonport/service/service_impl.go

```diff
@@ -0,0 +1,130 @@
+package service
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	datastore "github.com/stackrox/rox/central/processlisteningonport/datastore"
+	"github.com/stackrox/rox/central/role/resources"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/grpc/authz"
+	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
+	"github.com/stackrox/rox/pkg/grpc/authz/user"
+	"google.golang.org/grpc"
+	// "github.com/stackrox/rox/pkg/grpc/authz/allow"
+)
+
+var (
+	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
+		user.With(permissions.View(resources.ProcessListeningOnPort)): {
+			"/v1.ProcessesListeningOnPortsService/GetProcessesListeningOnPortsByNamespace",
+			"/v1.ProcessesListeningOnPortsService/GetProcessesListeningOnPortsByNamespaceAndDeployment",
+		},
+	})
+)
+
+type serviceImpl struct {
+	dataStore datastore.DataStore
+}
+
+// RegisterServiceServer registers this service with the given gRPC Server.
+func (s *serviceImpl) RegisterServiceServer(grpcServer *grpc.Server) {
+	v1.RegisterProcessesListeningOnPortsServiceServer(grpcServer, s)
+}
+
+// RegisterServiceHandler registers this service with the given gRPC Gateway endpoint.
+func (s *serviceImpl) RegisterServiceHandler(ctx context.Context, mux *runtime.ServeMux, conn *grpc.ClientConn) error {
+	return v1.RegisterProcessesListeningOnPortsServiceHandler(ctx, mux, conn)
+}
+
+// AuthFuncOverride specifies the auth criteria for this API.
+func (s *serviceImpl) AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
+	//// XXX: An anonymous access only for development
+	//  return ctx, allow.Anonymous().Authorized(ctx, fullMethodName)
+	return ctx, authorizer.Authorized(ctx, fullMethodName)
+}
+
+func (s *serviceImpl) GetProcessesListeningOnPortsByNamespace(ctx context.Context, req *v1.GetProcessesListeningOnPortsByNamespaceRequest) (*v1.GetProcessesListeningOnPortsWithDeploymentResponse, error) {
```

Comment:
Do we really need 2 endpoints or could we just use one and allow the deployment to be optional?  Just thinking out loud.  Also these 2 methods do very similar things so there may be some consolidation we can do and move some of it to functions.

Reply:
@JoukoVirtanen what do you think?

Reply:
We will go with one endpoint. I will soon create a PR to get rid of specifying the namespace.